packages/bundler-plugin-core/src/utils/Output.ts

```diff
@@ -25,28 +39,79 @@ class Output {
   assets?: Asset[];
   chunks?: Chunk[];
   modules?: Module[];
-  plugin?: {
+  private internalPlugin?: {
     name: string;
     version: string;
   };
 
   constructor(userOptions: NormalizedOptions) {
     this.version = "1";
     this.userOptions = userOptions;
+    this.internalOptions = {
+      frozenBundleName: false,
+      frozenPluginDetails: false,
+    };
   }
 
-  start(pluginName: string, pluginVersion: string) {
+  start() {
     this.builtAt = Date.now();
-    this.plugin = {
-      name: pluginName,
-      version: pluginVersion,
-    };
   }
 
   end() {
     this.duration = Date.now() - (this.builtAt ?? 0);
   }
 
+  setBundleName(bundleName: string, options?: SetPluginOptions) {
+    // if set to false, unfreeze the plugin details
+    if (typeof options?.frozen === "boolean" && !options.frozen) {
+      this.internalOptions.frozenBundleName = options.frozen;
```

Comment:
Ah okay, so because the process is only being ran once, and we need to share the Output class between both the Nuxt plugin and Vite plugin and for both client and server builds, we need some way to unfreeze the bundle name between those two runs so we can unlock it for the nuxt plugin, but then refreeze it so the vite plugin doesn't override it.

Reply:
So we'd end up with something like:

```ts
options.unlockBundleName()
options.setBundleName("...")
options.lockBundleName()
```